proxy/logging/LogFilter.h

```diff
@@ -382,6 +382,70 @@ LogFilterString::_checkCondition(OperatorFunction f, const char *field_value, si
   return retVal;
 }
 
+/*---------------------------------------------------------------------------
+ * find pattern from the query param
+ * 1) if pattern is not in the query param, return nullptr
+ * 2) if get the pattern in one param's value, seach again until it's in one param name, or nullptr if can't find it from param name
+---------------------------------------------------------------------------*/
+static const char *
+findPatternFromParamName(const char *lookup_query_param, const char *pattern)
+{
+  const char *pattern_in_query_param = strstr(lookup_query_param, pattern);
+  while (pattern_in_query_param) {
+    // wipe pattern in param name, need search again if find pattern in param value
+    const char *param_value_str = strchr(pattern_in_query_param, '=');
+    if (!param_value_str) {
+      // no "=" after pattern_in_query_param, means pattern_in_query_param is not in the param name, and no more param after it
+      pattern_in_query_param = nullptr;
+      break;
+    }
+    const char *param_name_str = strchr(pattern_in_query_param, '&');
+    if (param_name_str && param_value_str > param_name_str) {
+      //"=" is after "&" followd by pattern_in_query_param, means pattern_in_query_param is not in the param name
+      pattern_in_query_param = strstr(param_name_str, pattern);
+      continue;
+    }
+    // ensure pattern_in_query_param is in the param name now
+    break;
+  }
+  return pattern_in_query_param;
+}
+
+/*---------------------------------------------------------------------------
+ * replace param value whose name contains pattern with same count 'X' of original value str length
+---------------------------------------------------------------------------*/
+static void
+updatePatternForFieldValue(char **field, const char *pattern_str, int field_pos, char *buf_dest)
+{
+  int buf_dest_len = strlen(buf_dest);
+  char buf_dest_to_field[buf_dest_len + 1];
+  char *temp_text = buf_dest_to_field;
+  memcpy(temp_text, buf_dest, (pattern_str - buf_dest));
+  temp_text += (pattern_str - buf_dest);
+  const char *value_str = strchr(pattern_str, '=');
+  if (value_str) {
+    value_str++;
+    memcpy(temp_text, pattern_str, (value_str - pattern_str));
+    temp_text += (value_str - pattern_str);
+    const char *next_param_str = strchr(value_str, '&');
+    if (next_param_str) {
+      for (int i = 0; i < (next_param_str - value_str); i++)
+        temp_text[i] = 'X';
+      temp_text += (next_param_str - value_str);
+      memcpy(temp_text, next_param_str, ((buf_dest + buf_dest_len) - next_param_str));
+    } else {
+      for (int i = 0; i < ((buf_dest + buf_dest_len) - value_str); i++) {
+        temp_text[i] = 'X';
+      }
+    }
+  } else {
+    return;
+  }
+
+  buf_dest_to_field[buf_dest_len] = '\0';
+  strcpy(*field, buf_dest_to_field);
+}
+
```

Comment:
A general comment, it's better to add tests for the PR. We do have `Catch` for unit test and `Autest` for integration test available now for ATS development.

Reply:
Add autest og-filter.test.py in tests/gold_tests/logging/